packages/react-mongo/react-mongo.ts

```diff
@@ -0,0 +1,63 @@
+import { Meteor } from 'meteor/meteor'
+import { Mongo } from 'meteor/mongo'
+import { Tracker } from 'meteor/tracker'
+import { useEffect, useMemo, useReducer, useRef, DependencyList } from 'react'
+
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+const useSubscriptionClient = (factory: () => Meteor.SubscriptionHandle, deps: DependencyList = []) => {
+  const forceUpdate = useForceUpdate()
+  const subscription = useRef<Meteor.SubscriptionHandle>({
+    stop() {},
+    ready: () => false
+  })
+
+  useEffect(() => {
+    subscription.current = factory()
+    const computation = Tracker.autorun(() => {
+      if (subscription.current.ready()) forceUpdate()
+    })
+
+    return () => {
+      computation.stop()
+    }
+  }, deps)
+
+  return subscription.current
+}
+
+const useSubscriptionServer = (): Meteor.SubscriptionHandle => ({
+  stop() {},
+  ready() { return true }
+})
+
+export const useSubscription = (factory: () => Meteor.SubscriptionHandle, deps: DependencyList = []) => (
+  Meteor.isServer
+    ? useSubscriptionServer()
+    : useSubscriptionClient(factory, deps)
+)
+
+const useCursorClient = <T = any>(factory: () => Mongo.Cursor<T>, deps: DependencyList = []) => {
+  const cursor = useMemo<Mongo.Cursor<T>>(() => Tracker.nonreactive(factory), deps)
+  const forceUpdate = useForceUpdate()
+  useEffect(() => {
+    const observer = cursor.observeChanges({
+      added: forceUpdate,
+      changed: forceUpdate,
+      removed: forceUpdate
+    })
+    return () => {
+      observer.stop()
+    }
+  }, [cursor, ...deps])
```

Comment:
I think you are right. deps is used above for memoization with useMemo, that's probably enough.

Do you think `useMemo` is enough? There are cases where `useMemo` will dump it's memoized values. There's no danger of things leaking though, since if the cursor ref changes, things will be cleaned up in `useEffect`'s cleanup function, but we can make the reference more stable using `useRef` or something